### PR TITLE
Ambient sound performance improvements

### PR DESCRIPTION
**Pull request by ElectroJr**
Makes the client side ambient sound system perform slightly better. Its still pretty shit because it still periodically iterates through every entity currently visible on screen. Eventually it is probably worth just giving it its own client-side tree (e.g., as is done for sprites & lights).

### CODE REVIEW

File: Content.Client/Audio/AmbientSoundSystem.cs [hunk 1]
```diff
@@ -164,36 +175,37 @@ private void ClearSounds()
...
-                    Transform(x.Owner).Coordinates.TryDistance(EntityManager, coordinates, out var dist) ? dist * (x.Volume + 32) : float.MaxValue).ToList();
+                if (ambientComp.Sound is SoundPathSpecifier path)
+                    key = path.Path?.ToString() ?? string.Empty;
+                else
+                    key = ((SoundCollectionSpecifier) ambientComp.Sound).Collection ?? string.Empty;
+
+                var list = sourceDict.GetOrNew(key);
+
+                // wtf is this sorting even for? If its for prioritizing higher playing more audible sounds before others, shouldn't further away be counteracting higher volume?
+                list.Add(range * (ambientComp.Volume + 32), ambientComp);
```
Comment (ElectroJr) [inline Content.Client/Audio/AmbientSoundSystem.cs:208]: I'm not sure what the logic behind the previous sorting was, or where the magic "+32" comes from.

Comment (ElectroJr) [reply Content.Client/Audio/AmbientSoundSystem.cs:208]: also yeah there's a typo in the comment, but the comment should get resolved before merging anyways

Comment (moonheart08) [reply Content.Client/Audio/AmbientSoundSystem.cs:208]: i don't remember what the +32 is for but the idea is indeed to prioritize by volume. a further away but louder sound gets higher priority.

Comment (ElectroJr) [reply Content.Client/Audio/AmbientSoundSystem.cs]: Okay, I've just left it as is, though I did flip the sorting so it matches how it is currently sorted on master.

**Pull request merged**